bpf/lib/egress_policies.h

```diff
@@ -38,10 +43,9 @@ bool egress_gw_request_needs_redirect(struct iphdr *ip4, __u32 *tunnel_endpoint)
 	if (!egress_gw_policy)
 		return false;
 
-	/* If the gateway IP is 0.0.0.0 it means this is an excluded CIDR, so
-	 * skip redirection
-	 */
-	if (!egress_gw_policy->gateway_ip)
+	/* FIXME: no gateway traffic should be dropped in a future release */
```

Comment:
Almost there :slightly_smiling_face:

`K8sDatapathEgressGatewayTest tunnel disabled with endpointRoutes enabled egress gw policy with exclusion CIDRs` is failing as we need to handle this new case also in `egress_gw_snat_needed()` and `egress_gw_reply_needs_redirect()`

Reply:
Indeed I updated those with the new macro now, should be good hopefully :crossed_fingers: